### PR TITLE
Test scaffolding

### PR DESCRIPTION
**Pull request by markdorison**
Rebased these changes from PR #5 on top of latest changes from master:
* Extract common functions into util library.
* Extract purgeAll functionality.
* Added tests for purgeAll and util.
* Fix README heading levels.

In addition to #5, made the following change:
* Restructured file layout.

### CODE REVIEW

**Review by agarzola — APPROVED** [review]
🎉

File: package.json [hunk 1]
```diff
@@ -3,6 +3,9 @@
   "version": "0.4.1",
   "description": "Command line interface for the Fastly API.",
   "author": "Mark Dorison <mark@chromatichq.com>",
+  "contributors": [
+    "Alfonso Gómez-Arzola"
+  ],
```
Comment (agarzola) [inline package.json:8]: 😍

File: src/commands/purge-all.test.js [hunk 2]
```diff
@@ -40,7 +32,8 @@ describe(`purge-all`, () => {
 
   it(`should instantiate Fastly with API key`, (done) => {
     testSubject(options, Fastly, util);
-    Fastly.getCalls(0)[0].args.should.deepEqual([options.apikey]);
+    Fastly.should.be.called();
```
Comment (agarzola) [inline src/commands/purge-all.test.js]: Do you think there’s value in checking that it’s `.calledOnce()` here?

**Review by agarzola — APPROVED** [review]
Other than the question I just posted (which isn’t really a request for change), this looks good.

File: src/commands/purge-all.test.js (hunk 2 above)
Comment (markdorison) [reply src/commands/purge-all.test.js]: The more specific the better, I would say. On it.

**Pull request merged**